src/audio/codec_adapter/codec_adapter.c

```diff
@@ -703,6 +712,7 @@ void codec_adapter_free(struct comp_dev *dev)
 		buffer_free(buffer);
 	}
 
+	rfree(mod->stream_params);
```

Comment:
I think we should free params in reset instead of free. With static pipelines, we'd never ever free the module unless we enter D3

Reply:
Fixed.